### PR TITLE
[refactor] move grpc server logic into sources/util

### PR DESCRIPTION
**Pull request by caibirdme**
Since sources/vector is actually a grpc server, and there're maybe a lot of other sources based on the grpc(like opentelemetry), so it'd better to move those logic into sources/util/grpc to enhance its resuability.
New sources like opentelemetry, could just need to new its proto:Service, like:
```rust
let service = proto::Server::new(Service {
        pipeline: cx.out,
        acknowledgements,
    })
    .accept_gzip();
run_grpc_server(addr, tls_setting, service, shutdown).map_err(|x| error!("xxx"))

       
```

### CODE REVIEW

**Review by tobz — APPROVED** [review]
Generally speaking, this looks fine to me. 👍🏻 

Just have to wait for CI to pass.

File: src/sources/util/mod.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@ mod encoding_config;
 pub mod finalizer;
 #[cfg(all(unix, feature = "sources-dnstap"))]
 pub mod framestream;
+pub mod grpc;
```
Comment (tobz) [inline src/sources/util/mod.rs:15]: Looks like CI highlighted the fact that this gets pulled in even when nothing is using it, which results in errors.

You can pull it in conditionally pretty simply:
```suggestion
#[cfg(feature = "sources-vector")]
pub mod grpc;

**Review by tobz — CHANGES_REQUESTED** [review]
Woopsies, looks like I missed a spot when reviewing.

File: src/sources/util/mod.rs (hunk 1 above)
Comment (caibirdme) [reply src/sources/util/mod.rs:16]: Done

**Review by StephenWakely — APPROVED** [review]

**Pull request merged**